packages/cli/src/logging/logger.service.ts

```diff
@@ -61,7 +62,7 @@ export class Logger {
 	/** Create a logger that injects the given scopes into its log metadata. */
 	scoped(scopes: LogScope | LogScope[]) {
 		scopes = Array.isArray(scopes) ? scopes : [scopes];
-		const scopedLogger = new Logger(this.globalConfig, this.instanceSettings);
+		const scopedLogger = new Logger(this.globalConfig, this.instanceSettings, { isRoot: false });
```

Comment:
do you think it might make sense to move the minimal amount of code needed for scoped loggers to a separate class, instead of reusing this class for two separate purposes?

Reply:
Interesting, let me try this and report back

Reply:
Tried this quickly [here](https://github.com/n8n-io/n8n/compare/separate-scoped-logger?expand=1). The scoped logger relies heavily on the existing logger, so 

- if the scoped logger doesn't extend from the existing logger, we'd have to extract quite a bit of deeply integrated logic or duplicate quite a bit, also using a separate scoped logger like this introduces a circular dependency `ReferenceError: Circular dependency: TargetServer. Index: 0` that I haven't spent time debugging yet
- if the scoped logger does extend from the existing logger to reuse logic and avoid extracting most of it, then we're back at wanting a way to avoid initializing the proxy on every constructor call, which is what this PR is addressing

Let me know if worth investing more time into this

Reply:
does the scoped logger also need to use winston to write logs to the file?

also, if you prefer, we can merge this as it is, and I can have a look at splitting this in a separate PR.

Reply:
> does the scoped logger also need to use winston to write logs to the file?

Yes, the scoped logger does everything the existing logger does, with the only addition that it auto-injects the scope string into the metadata of every log it emits, no matter the transport, etc.

> also, if you prefer, we can merge this as it is, and I can have a look at splitting this in a separate PR.

Let's do this and revisit. I'd like to fix this minor bug quickly.